src/main/java/org/spongepowered/common/block/SpongeBlockSnapshot.java

```diff
@@ -365,6 +365,26 @@ public int getUpdateFlag() {
     }
 
     @Override
+    public int hashCode() {
+        return Objects.hashCode(this.worldUniqueId, this.pos, this.blockState, this.extraData);
```

Comment:
It does due to the contains check in EntityLightningBolt for the block transaction as the objects are not necessarily the same I needed to implement this.